vng_api_common/validators.py

```diff
@@ -22,6 +21,89 @@
 WORD_REGEX = re.compile(r"[\w\-]+$", re.ASCII)
 
 
+class BaseValidator:
+    """
+    Validator base class that performs common validation logic.
+    Digit check, length, and optional 11-proof check.
+    """
+
+    error_messages = {
+        "isdigit": _("Voer een numerieke waarde in"),
+        "length": _(
+            "De lengte van de waarde moet gelijk zijn aan een van deze waarden: %(list_size)s"
+        ),
+        "11proefnumber": _("Ongeldige code"),
+    }
+
+    def __init__(
+        self,
+        value: str,
+        list_size: list[int] | None = [],
+        check_11proefnumber: bool = False,
+    ):
+        self.value = value
+        self.list_size = list_size
+        self.check_11proefnumber = check_11proefnumber
+
+    def validate_isdigit(self) -> None:
+        """Validates that the value contains only digits."""
+        if not self.value.isdigit():
+            raise ValidationError(self.error_messages["isdigit"], code="only-digits")
+
+    def validate_length(self) -> None:
+        """Validates that the length of the value is within the allowed sizes."""
+        if len(self.value) not in self.list_size:
+            raise ValidationError(
+                self.error_messages["length"] % {"list_size": self.list_size},
+                code="invalid-length",
+            )
+
+    def validate_11proefnumber(self) -> None:
+        """Validates the value based on the 11-proof check."""
+        total = 0
+        for multiplier, char in enumerate(reversed(self.value), start=1):
+            if multiplier == 1:
+                total += -multiplier * int(char)
+            else:
+                total += multiplier * int(char)
+
+        if total % 11 != 0:
+            raise ValidationError(
+                self.error_messages["11proefnumber"], code="invalid-code"
```

Comment:
Let's keep this the same for backward compatibility
```suggestion
                self.error_messages["11proefnumber"], code="invalid"
```